src/app/activity/bounty/Submit.ts

```diff
@@ -1,45 +1,130 @@
+        // Since card may have been in a DM, guild might not be populated in the request
+        if (request.guildId === undefined || request.guildId === null) {
+            request.guildId = getDbResult.dbBountyResult.customerId;
+        }
+        const submittedByUser = await DiscordUtils.getGuildMemberFromUserId(request.userId, request.guildId);
+        const createdByUser: GuildMember = await submittedByUser.guild.members.fetch(getDbResult.dbBountyResult.createdBy.discordId);
+        Log.info(`${request.bountyId} bounty submitted by ${submittedByUser.user.tag}`);
+    
+        await writeDbHandler(request, submittedByUser);
+    
+        const cardMessage = await BountyUtils.canonicalCard(getDbResult.dbBountyResult._id, request.activity);
+    
+        let creatorSubmitDM = `Please reach out to <@${submittedByUser.user.id}>. They are ready for bounty review`
+    
+        if (request.notes) {
+            creatorSubmitDM += `\nNotes included in submission:\n${request.notes}`
+        }
+        await DiscordUtils.activityNotification(creatorSubmitDM, createdByUser, cardMessage.url);
+        await DiscordUtils.activityResponse(request.commandContext, request.buttonInteraction, `Bounty in review! Expect a message from <@${createdByUser.id}>.`, cardMessage.url);
+        return;
+
+    }
+
+    // Callback for the slash modal version
+    const modalCallback = async (modalContext: ModalInteractionContext, request: SubmitRequest) => {
+        await modalContext.defer(true);
+        request.notes = modalContext.values.submission_notes;
+        await finishSubmit(request, modalContext);
+    };
+
+    // Call the modal. For slash command (slash-create), call the callback. For button interaction (discord.js), wait for submit and return. 
+
+    if (fromSlash) {
+        try {
+            await request.commandContext.sendModal(modal as unknown as scModalOptions,async (mctx) => { await modalCallback(mctx, request) });
+        } catch(e) {
+            Log.error(e.message);
+            throw new RuntimeError(e);
+        }
+        return;
+    } else {
+        const crypto = require('crypto');
+        const uuid = crypto.randomUUID();
+        try {
+            await request.buttonInteraction.showModal(Object.assign(modal, {customId: uuid}) as unknown as ModalOptions);
+        } catch(e) {
+            Log.error(e.message);
+            throw new RuntimeError(e);
+        }
+        const submittedInteraction = await request.buttonInteraction.awaitModalSubmit({
+            time: 60000,
+            filter: i => (i.user.id === request.userId) && (i.customId === uuid),
+            }).catch(e => {
+                Log.info(`<@${request.userId}> had a modal error ${e.message}`);
+                // Most likely a modal form timeout error
+                throw new ModalTimeoutError(e);
+            }) as ModalSubmitInteraction;
+        request.notes = submittedInteraction.components[0].components[0].value;
```

Comment:
Should we define an enum as to be explicit about the ordering of fields we define in the modal?